grid2op/Environment/environment.py

```diff
@@ -86,6 +86,7 @@ def __init__(
         parameters,
         name="unknown",
         n_busbar : N_BUSBAR_PER_SUB_TYPING=DEFAULT_N_BUSBAR_PER_SUB,
+        allow_shedding:bool=DEFAULT_ALLOW_SHEDDING,
```

Comment:
when you have another attribute like this one in the constructor, you need also to implement:
- `get_kwargs`
- `get_params_for_runner` which in turn caused you to change the Runner class with the methods (from Runner): 
   - `__init__` 
- `init_obj_from_kwargs` and modify the `MaskedEnvironment` and `TimedOutEnvironment` classes
- the `ObsEnv` and the `ObservationSpace` also, to propagate this feature to the `obs.simulate` and co
- maybe in this case the `Simulator` too 
- you might also need to adapt the `_custom_deepcopy_for_copy` depending on whether or not you added an attribute to the Environment class (a